s3-public/s3.tf

```diff
@@ -1,24 +1,34 @@
 resource "aws_s3_bucket" "main" {
   bucket = var.bucket_name
-  acl    = "public-read"
+  tags = {
+    Name        = var.bucket_name
+    Project     = var.project
+    Environment = var.environment
+  }
+}
 
-  versioning {
-    enabled = var.versioning
+resource "aws_s3_bucket_versioning" "main-bucket-versioning" {
+  bucket = aws_s3_bucket.main.id
+  versioning_configuration {
+    status = var.versioning ? "Enabled" : "Disabled"
   }
+}
 
-  # Move data to a cheaper storage class after a period of time
-  lifecycle_rule {
-    enabled = var.primary_storage_class_retention == 0 ? false : false
+resource "aws_s3_bucket_acl" "main-bucket-data-acl" {
+  bucket = aws_s3_bucket.main.id
+  acl    = "public-read"
+}
+
+# Move data to a cheaper storage class after a period of time
+resource "aws_s3_bucket_lifecycle_configuration" "main-bucket-lifecycle-rule" {
+  bucket = aws_s3_bucket.main.id
 
+  rule {
+    id     = "primary-storage-class-retention"
+    status = var.primary_storage_class_retention == 0 ? "Disabled" : "Enabled"
     noncurrent_version_transition {
-      days          = var.primary_storage_class_retention == 0 ? 365 : var.primary_storage_class_retention
-      storage_class = "STANDARD_IA"
+      noncurrent_days = var.primary_storage_class_retention == 0 ? 365 : var.primary_storage_class_retention
```

Comment:
Params for noncurrent_version_transition

https://registry.terraform.io/providers/hashicorp/aws/3.75.0/docs/resources/s3_bucket_lifecycle_configuration#noncurrent_version_transition